### PR TITLE
WIP: Add PatchMapWithMove2

### PR DESCRIPTION
**Pull request by Ericson2314**
OK so here's what needs to happen:

1. Fix hlint *done*

2. ~~Reflex.Patch.MapWithMove is written in terms of `Reflex.Patch.MapWithMove2`. We use `Proxy a` and a blank patch for `a` to do this. Add an instance for `Proxy a` if needed.~~

3. ~~`Reflex.Patch.MapWithMove` is a deprecated module, `Reflex.Patch.MapWithMove2` is renamed `Reflex.Patch.MapWithMove.New`. Sometime later `Reflex.Patch.MapWithMove.New` becomes `Reflex.Patch.MapWithMove` and the old one is deleted.~~

OK the patching part of this is moved to https://github.com/reflex-frp/patch/pull/1/files . We should make reflex use that library, and then this PR use that PR.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -42,8 +42,7 @@
 
 ## 0.6.2.0
 
-* Fix `holdDyn` so that it is lazy in its event argument  
-  These produce `DMap`s  whose values needn't be `Identity`.
+* Add `Reflex.Query.Base.mapQueryT`. See that module for documentation
```
Comment (3noch) [inline ChangeLog.md]: Was this change log entry for a past release wrong?

File: ChangeLog.md [hunk 2]
```diff
@@ -42,8 +42,7 @@
...
 * Stop using the now-deprecated `*Tag` classes (e.g., `ShowTag`).
 * Fix `holdDyn` so that it is lazy in its event argument.
```
Comment (alexfmpe) [inline ChangeLog.md]: It was duplicated here, it seems

Comment (alexfmpe) [reply ChangeLog.md]: Hmm only half of it though?

Comment (Ericson2314) [reply ChangeLog.md]: We should fix that in another PR. In the meantime, I moved the new entry for this PR's changes to the top where it belongs.

**Comment by 3noch** [comment]
It appears CI fails due to hlint

**Comment by Ericson2314** [comment]
Oh good, that makes this easier.

**Comment by Ericson2314** [comment]
OK so here's what needs to happen:

1. Reflex.Patch.MapWithMove is written in terms of `Reflex.Patch.MapWithMove`. We use `Proxy a` an a blank patch for `a` to do this. Add an instance for `Proxy a` if needed.

2. `Reflex.Patch.MapWithMove` is a deprecated module`, `Reflex.Patch.MapWithMove2` is renamed `Reflex.Patch.MapWithMove.New`. Sometime later `Reflex.Patch.MapWithMove.New` becomes `Reflex.Patch.MapWithMove` and the old one is deleted.

**Comment by Ericson2314** [comment]
With the non-patch parts already merged (`mapQueryT`) the rest is moved to https://github.com/reflex-frp/patch/pull/1

**Pull request closed without merging**